### DISCUSSION TITLE
How to inspect elements on WebView2 apps?

### DISCUSSION THREAD

**Discussion by italodeverdade** [MicrosoftEdge/WebView2Feedback#4080] [Q&A]
Hello! I'm integrating Microsoft Teams' new app into my work application that currently supports the old Teams app. While moving to the "new Teams" with WebView2 I realize that accessibility functions are not working now. Here is a snippet of a code that works in the previous version.

```cpp

// helper functions class
CComPtr<IUIAutomationElement> WebView2Helper::GetElementByAutomationId(HWND hwnd, const std::string& id) {
  if (FAILED(CoInitializeEx(NULL, COINIT_MULTITHREADED))) {
    return nullptr;
  }
  CComPtr<IUIAutomation> automation;
  if (FAILED(CoCreateInstance(CLSID_CUIAutomation, nullptr, CLSCTX_INPROC_SERVER, IID_IUIAutomation,
                              (void**)&automation))) {
    return nullptr;
  }

  VARIANT property;
  VariantInit(&property);
  property.vt = VT_BSTR ;
  property.bstrVal = SysAllocString(std::wstring(id.begin(), id.end()).c_str());


  CComPtr<IUIAutomationCondition> condition;
  if (FAILED(
          automation->CreatePropertyCondition(UIA_AutomationIdPropertyId, property, &condition))) {
    return nullptr;
  }

  CComPtr<IUIAutomationElement> root_element;
  if (FAILED(automation->ElementFromHandle(hwnd, &root_element))) {
    return nullptr;
  }

  CComPtr<IUIAutomationElement> target_element;
  if (FAILED(root_element->FindFirst(TreeScope_Descendants, condition, &target_element))) {
    return nullptr;
  }

  if(!target_element){
    std::cout << "[WebView2Helper] Element not found" << std::endl;
  }

  CoUninitialize();

  return target_element;
}

// main file
void TeamsHelper::GetData(){
   auto w = GetDesktopWindow();
   auto element = WebView2Helper::GetElementByAutomationId(w, "call-duration-custom");
}

```

![image](https://github.com/MicrosoftEdge/WebView2Feedback/assets/4932642/6c66da8e-6de4-4662-bb8a-f1c9aec836e5)

I tried to find in docs or any other repo on GitHub / internet about integrating correctly UIA with WebView2 but right now I'm without any clue. I appreciate any help :)

**Comment by victorhuangwq** [comment]
Hi! @tochukwuIbeEkeocha or @jennifer-jacobs could you help out with this issue?

**Reply by italodeverdade** [reply]
Any updates here @victorhuangwq? 😢

**Comment by jennifer-nejad** [comment]
Hi! To make sure I'm understanding your scenario correctly, you're wrapping the teams app inside your app? What framework is your app built on (WinUI2 or WinUI3, Win32, WinForms, WPF)?

Is your method able to find any element in the tree (like one of the top level ones in the Accessibility Insights screenshot you sent, such as pane "Meeting with tldv bot | Microsoft Teams")? Or can it not find any element in the tree?
